lib/Application.js

```diff
@@ -142,9 +145,21 @@ Application.prototype = {
   },
 
   /**
+   * Handles the DOM captured by the plugin.
+   *
+   * @param {Object} data
+   *   an object containing the dom and url attributes.
+   */
+  _handleDOMCapture(data) {
+    //TO-DO: capture data!
+  },
+
+  /**
    * Sets up various listeners for the pages
    */
   _setupListeners() {
+    this.DOMFetcher.onCapture(this._handleDOMCapture);
```

Comment:
I think it makes sense to only handle the `init`, and `uninit` in `Application`. The details of what to do with the DOM should be contained elsewhere.